src/app/components/upload-annotations/details/details.component.html

```diff
@@ -0,0 +1,239 @@
+            size="sm"
+          ></baw-loading>
+
+          <ng-template #audioRecordingTemplate>
+            <a [bawUrl]="value.audioRecording.viewUrl">
+              {{ value.audioRecording.id }}
+            </a>
+          </ng-template>
+        </ng-template>
+      </ngx-datatable-column>
+
+      <ngx-datatable-column prop="createdAt">
+        <ng-template let-column="column" ngx-datatable-header-template>
+          Created At
+        </ng-template>
+
+        <ng-template let-value="value" ngx-datatable-cell-template>
+          <span>
+            {{ value | dateTime : { includeTime: true, localTime: true } }}
+          </span>
+        </ng-template>
+      </ngx-datatable-column>
+
+      <ngx-datatable-column prop="">
+        <ng-template let-column="column" ngx-datatable-header-template>
+          Associated Tags
+        </ng-template>
+
+        <ng-template let-value="value" ngx-datatable-cell-template>
+          <baw-loading
+            *ngIf="value.tags | isUnresolved; else tagsTemplate"
+            size="sm"
+          ></baw-loading>
+
+          <ng-template #tagsTemplate>
+            <baw-inline-list
+              [items]="value.tags"
+              [itemKey]="'text'"
+              [emptyTemplate]="noAssociatedTagsTemplate"
+            >
+            </baw-inline-list>
+          </ng-template>
+        </ng-template>
+      </ngx-datatable-column>
+    </ngx-datatable>
+  </section>
+
+  <section>
+    <h2>Files</h2>
+
+    <table
+      *ngIf="audioEventImport.files.length > 0; else noFilesTemplate"
+      class="table"
+    >
+      <thead>
+        <tr>
+          <th scope="col">File Name</th>
+          <th scope="col">Date Imported</th>
+          <th scope="col">Associated Tags</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        <tr *ngFor="let file of audioEventImport.files" scope="row">
+          <td>{{ file.name }}</td>
+          <td>
+            <span>
+              {{ file.importedAt | dateTime : { includeTime: true, localTime: true } }}
+            </span>
+          </td>
+          <td>
+            <baw-inline-list
+              [items]="file.additionalTagModels"
+              [itemKey]="'text'"
+              [emptyTemplate]="noAssociatedTagsTemplate"
+            ></baw-inline-list>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </section>
+</div>
+
+<section>
+  <h2>Upload new annotations</h2>
+
+  <div class="annotationsUploadInput">
+    <div class="upload-groups">
+      <ng-container *ngFor="let eventGroup of uploadBuffer; let index = index">
+        <ng-template
+          [ngTemplateOutlet]="uploadGroupTemplate"
+          [ngTemplateOutletContext]="{ model: eventGroup, index: index }"
+        ></ng-template>
+      </ng-container>
+    </div>
+  </div>
+
+  <span
+    [ngbTooltip]="
+      isBufferValid()
+        ? 'Upload all new annotations'
+        : 'Please fill out all required fields or fix errors'
+    "
```

Comment:
multiline attributes always seem icky to me. code smell?

also what is a buffer in this context?

Reply:
A buffer is all the "upload groups" pending for upload. Each of which unique files (multiple), errors, associated tags, and additional tags.

([view model](https://github.com/QutEcoacoustics/workbench-client/blob/annotations-upload/src/app/components/upload-annotations/details/details.component.ts#L45-L58))